EventFilter/EcalRawToDigi/src/UnpackGPU.cu

```diff
@@ -0,0 +1,476 @@
+        iphi += 5 * (ElectronicsIdGPU::kTowersInPhi - ((tower - 1) % ElectronicsIdGPU::kTowersInPhi) - 1);
+    }
+
+    bool RightTower = right_tower_for_eb(tower);
+    int ieta = 5 * ((tower - 1) / ElectronicsIdGPU::kTowersInPhi) + 1;
+    if (RightTower) {
+        ieta += (strip - 1);
+        if (strip % 2 == 1) {
+            if (EBMinus)
+                iphi += (xtal - 1) + 1;
+            else
+                iphi += (4 - (xtal - 1)) + 1;
+        } else {
+            if (EBMinus)
+                iphi += (4 - (xtal - 1)) + 1;
+            else
+                iphi += (xtal - 1) + 1;
+        }
+    } else {
+        ieta += 4 - (strip - 1);
+        if (strip % 2 == 1) {
+            if (EBMinus)
+                iphi += (4 - (xtal - 1)) + 1;
+            else
+                iphi += (xtal - 1) + 1;
+        } else {
+            if (EBMinus)
+                iphi += (xtal - 1) + 1;
+            else
+                iphi += (4 - (xtal - 1)) + 1;
+        }
+    }
+    
+    if (zside_for_eb(eid) < 0)
+        ieta = -ieta;
+
+    DetId did{DetId::Ecal, EcalBarrel};
+    return  did.rawId() |  
+        ((ieta > 0) 
+            ? (0x10000 | (ieta << 9)) 
+            : ((-ieta) << 9)) | (iphi & 0x1FF);
+}
+
+__forceinline__ __device__
+int adc(uint16_t sample) { return sample & 0xfff; }
+__forceinline__ __device__
+int gainId(uint16_t sample) { return (sample>>12) & 0x3; }
+
+template<int NTHREADS>
+__global__
+void kernel_unpack_test(
+        unsigned char const* __restrict__ data,
+        uint32_t const* __restrict__ offsets,
+        int const* __restrict__ feds,
+        uint16_t *samplesEB,
+        uint16_t *samplesEE,
+        uint32_t *idsEB,
+        uint32_t *idsEE,
+        uint32_t *pChannelsCounterEBEE,
+        uint32_t const* eid2did,
+        uint32_t const nbytesTotal) {
+    // indices
+    auto const ifed = blockIdx.x;
+
+    // FIXME: use only the very first fed
+    //if (ifed!=10) return;
+
+    // offset in bytes
+    auto const offset = offsets[ifed];
+    // fed id
+    auto const fed = feds[ifed];
+    auto const isBarrel = is_barrel(static_cast<uint8_t>(fed - 600));
+    // size
+    auto const size = ifed==gridDim.x-1 ? nbytesTotal - offset : offsets[ifed+1] - offset;
+    auto *samples = isBarrel ? samplesEB : samplesEE;
+    auto *ids = isBarrel ? idsEB : idsEE;
+    auto *pChannelsCounter = isBarrel 
+        ? &pChannelsCounterEBEE[0] 
+        : &pChannelsCounterEBEE[1];
+
+    // FIXME: debugging
+    //printf("ifed = %u fed = %d offset = %u size = %u\n", ifed, fed, offset, size);
+
+    // offset to the right raw buffer
+    uint64_t const* buffer = reinterpret_cast<uint64_t const*>(data + offset);
+
+    // dump first 3 bits for each 64-bit word
+    //print_first3bits(buffer, size / 8);
+
+    //
+    // fed header
+    //
+    //print_raw_buffer(reinterpret_cast<uint8_t const*>(buffer), 8);
+    //printf("\n");
+    auto const fed_header = buffer[0];
```

Comment:
From here on there are plenty of unused variables and commented out `printf` statements.
Is it OK to clean it all up ?